src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -254,28 +267,65 @@ private boolean isWithinAreasWithExcludedAmenityTags(final Edge edge)
     }
 
     /**
-     * Checks if an {@link Edge} is of type service
+     * Checks if an {@link Edge} is connected to any edge that is a pedestrian navigable highway
      * 
      * @param edge
      *            any edge
-     * @return true if the highway classification of the edge is of type service and it has a
-     *         service tag
+     * @return true if the edge has connection to pedestrian navigable highways
      */
-    private boolean isServiceRoad(final Edge edge)
+    private boolean isConnectedToPedestrianNavigableHighway(final Edge edge)
     {
-        return Validators.isOfType(edge, HighwayTag.class, HighwayTag.SERVICE)
-                && Validators.hasValuesFor(edge, ServiceTag.class);
+        return edge.connectedEdges().stream().anyMatch(HighwayTag::isPedestrianNavigableHighway);
     }
 
     /**
-     * Checks if an {@link Edge} is connected to any edge that is a pedestrian navigable highway
-     * 
+     * Checks if the edge is within or intersects airport polygon. Airport polygon is an atlas area
+     * with {@link AerowayTag} tag.
+     *
      * @param edge
      *            any edge
-     * @return true if the edge has connection to pedestrian navigable highways
+     * @return true if the edge is within or intersects airport polygon
      */
-    private boolean isConnectedToPedestrianNavigableHighway(final Edge edge)
+    private boolean intersectsAirportOrBuilding(final Edge edge)
     {
-        return edge.connectedEdges().stream().anyMatch(HighwayTag::isPedestrianNavigableHighway);
+        return StreamSupport
+                .stream(edge.getAtlas()
+                        .areasIntersecting(edge.bounds(),
+                                area -> Validators.hasValuesFor(area, AmenityTag.class)
+                                        || BuildingTag.isBuilding(area)
+                                        || Validators.hasValuesFor(area, AerowayTag.class))
+                        .spliterator(), false)
+                .anyMatch(area -> area.asPolygon().overlaps(edge.asPolyLine()));
+    }
+
+    /**
+     * Checks if the edge is car navigable in terms of {@link MotorVehicleTag}, {@link MotorcarTag}
+     * and {@link VehicleTag}. Edge is navigable if 1) MotorVehicleTag, MotorcarTag and VehicleTag
```

Comment:
So we're not getting any FP for edges that have an absent  MotorVehicleTag, MotorcarTag and VehicleTag ?

Reply:
@kkonishi2 is doing a final round of analysis on the pre and post enhancement to be 100% sure about this.